The Annotated Transformer.md

```diff
@@ -0,0 +1,1424 @@
+---
+jupyter:
+  jupytext:
+    formats: ipynb,md
+    text_representation:
+      extension: .md
+      format_name: markdown
+      format_version: '1.3'
+      jupytext_version: 1.11.4
+  kernelspec:
+    display_name: Python 3
+    language: python
+    name: python3
+---
+
+```python id="K5Rh7ZCJTsp4"
+from IPython.display import Image
+Image(filename='images/aiayn.png')
+```
+
+<!-- #region id="SX7UC-8jTsp7" -->
+The Transformer from ["Attention is All You Need"](https://arxiv.org/abs/1706.03762) has been on a lot of people's minds over the last year. Besides producing major improvements in translation quality, it provides a new architecture for many other NLP tasks. The paper itself is very clearly written, but the conventional wisdom has been that it is quite difficult to implement correctly. 
+
+In this post I present an "annotated" version of the paper in the form of a line-by-line implementation. I have reordered and deleted some sections from the original paper and added comments throughout. This document itself is a working notebook, and should be a completely usable implementation. In total there are 400 lines of library code which can process 27,000 tokens per second on 4 GPUs. 
+
+To follow along you will first need to install [PyTorch](http://pytorch.org/). The complete notebook is also available on [github](https://github.com/harvardnlp/annotated-transformer) or on Google [Colab](https://drive.google.com/file/d/1xQXSv6mtAOLXxEMi8RvaW8TW-7bvYBDF/view?usp=sharing). 
+
+Note this is merely a starting point for researchers and interested developers. The code here is based heavily on our [OpenNMT](http://opennmt.net) packages. (If helpful feel free to [cite](#conclusion).) For other full-sevice implementations of the model check-out [Tensor2Tensor](https://github.com/tensorflow/tensor2tensor) (tensorflow) and [Sockeye](https://github.com/awslabs/sockeye) (mxnet).
+
+- Alexander Rush ([@harvardnlp](https://twitter.com/harvardnlp) or srush@seas.harvard.edu)
+
+<!-- #endregion -->
+
+<!-- #region id="BhmOhn9lTsp8" -->
+# Prelims
+<!-- #endregion -->
+
+```python id="NwClcbH6Tsp8"
+# !conda install -c pytorch torchtext spacy altair
+```
+
+```python id="hin0sxaITsp9" tags=[]
+# !pip install http://download.pytorch.org/whl/cu80/torch-0.3.0.post4-cp36-cp36m-linux_x86_64.whl numpy matplotlib spacy torchtext seaborn 
+```
+
+```python id="v1-1MX6oTsp9"
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
```

Comment:
Let's remove this import and just import the needed functions. (Don't love the capital)